nameko_vault/__init__.py

```diff
@@ -65,3 +68,40 @@ def delete_metadata_and_all_versions_kv_secret(
             path=path,
         )
         return response.status_code
+
+
+class EnvLoaderProvider(DependencyProvider):
+    """
+    Starting from a `mount_point` and a `path`, get all environment variables
+    from Hashicorp Vault and inject those vars into the environment.
+    """
+
+    def __init__(self, mount_point: str, path: str):
+        self.mount_point = mount_point
+        self.path = path
+
+    def setup(self):
+        logging.info(
+            f"Importing env vars from Vault, mount_poit: '{self.mount_point}',"
+            f" path: '{self.path}'"
+        )
+        url = os.getenv("VAULT_URL")
+        token = os.getenv("VAULT_TOKEN")
+
+        client = hvac.Client(url=url)
+
+        if client.is_authenticated() is False:
+            client.token = token
+
+        env_vars = client.secrets.kv.read_secret_version(
+            mount_point=self.mount_point, path=self.path
+        )["data"]["data"]
+
+        logging.info("Vars loaded from Vault:")
+        for key, value in env_vars.items():
+            print(value)
+            logging.info(f"{key}: {value}")
+            os.environ[key] = value
+            print(os.getenv("VAR1"))
```

Comment:
Why use print instead logging like before?

Reply:
I forgot to delete these prints XD Sorry!